packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -104,8 +105,28 @@ export type SdkType =
 export interface SdkBuiltInType extends SdkTypeBase {
   kind: SdkBuiltInKinds;
   encode: string;
+  name: string;
+  tspNamespace?: string;
+  baseType?: SdkBuiltInType;
```

Comment:
what does the `baseType` give us? Won't the base scalar type be the `kind`?

Reply:
base gives us the `B` part in `scalar A extends B`.
This might be important in a case like this:
```typespec
// this is in azure core
scalar azureLocation extends string;

// this is in service spec
scalar computeLocation extends azureLocation;
```
if we do not have the `base`, in the result we will never have anything related with `azureLocation`.
More detailed information for this could be found in the PR description, please take a look as well

Reply:
I'm still not sure we need `baseType`, at least to start with. What does knowing that it extends from `azureLocation` give you? I'd prefer if we don't add it unless it's absolutely necessary, since `baseType` is also confusing in this context. To me, the `baseType` would eventually be string here, bc `string` is the true underlying base type of both `azureLocation` and `compulteLocation`

Reply:
the value of `baseType` here is that in the above case, we want to generate `AzureLocation` for this `computeLocation`, because the `extends` means "is a", we should not generate a `string` in this case.
Therefore we need the base type to let us know this information, otherwise we just lose the information of azure location here.